api/src/pcapi/connectors/ems.py

```diff
@@ -1,73 +1,130 @@
 
-    if api_response.status_code != 200:
-        raise EMSAPIException(f"Error getting EMS API movie poster {image_url} with code {api_response.status_code}")
+    def _build_headers(self) -> dict[str, str]:
+        """
+        Mandatory header to go through EMS API
+        """
+        return {"Source": settings.EMS_API_BOOKING_HEADER, "Content-Type": "application/json"}
 
-    return api_response.content
+    def _build_url(self, endpoint: str, payload: dict) -> str:
+        """
+        Build the url for EMS booking API according to their specifications.
 
+        The payload that is going to be sent is also hashed with sha512 and a secret key
+        and will be concatenate to the url we are calling.
 
-def _check_response_is_ok(response: requests.Response) -> None:
-    if response.status_code >= 400:
-        reason = _extract_reason_from_response(response)
-        message = _extract_message_from_response(response)
-        error_message = _filter_credentials(reason)
-        raise EMSAPIException(f"Error on EMS API call {error_message} - {message}")
+        Args:
+            endpoint (str): Endpoint we want to call (SEANCE/, VENTE/, ANNULATION/, ...)
+            payload (dict): Payload that is going to be sent
 
+        Returns:
+            url (str): Something like
+                https://EMS_API_BOOKING_URL/ENDPOINT/digest/
 
-def _extract_reason_from_response(response: requests.Response) -> str:
-    if isinstance(response.reason, bytes):
-        try:
-            reason = response.reason.decode("utf-8")
-        except UnicodeDecodeError:
-            reason = response.reason.decode("iso-8859-1")
-    else:
-        reason = response.reason
-    return reason
+        """
+        # We can't just do str(payload) here
+        # Because on EMS side they will hash our payload as well
+        # and hash(str(payload)) != hash(json.dumps(payload))
+        # (because '{"key": "value"}' is not "{'key': 'value'}")
+        # So we need to hash exactly what we are sending
+        # which are data as json format
+        payload_stringify = json.dumps(payload)
 
+        secret_key = settings.EMS_API_BOOKING_SECRET_KEY
+        base_url = settings.EMS_API_BOOKING_URL + endpoint
 
-def _extract_message_from_response(response: requests.Response) -> str:
-    try:
-        content = response.json()
-        message = content.get("message", "")
-    except json.JSONDecodeError:
-        message = response.content
-    return message
+        h = hmac.new(secret_key.encode(), digestmod=self.digest_mode)
+        h.update(payload_stringify.encode())
 
+        digest = h.hexdigest()
 
-def _filter_credentials(error_message: str) -> str:
-    credentials = f"{settings.EMS_API_USER}:{settings.EMS_API_PASSWORD}@"
-    error_message = error_message.replace(credentials, "")
-    return error_message
+        return base_url + digest
```

Comment:
```suggestion
        return f"{settings.EMS_API_BOOKING_URL}/{endpoint}/{digest}"
```
On pourrait utiliser des `str` d'endpoint sans trailing `/` afin d'éviter les oublis, comme pour les autres providers ciné, par exemple `VENTE`.